wip-encode-error.md

```diff
@@ -0,0 +1,92 @@
+<pre>
+  WIP: WIP-encode-error
+  Layer: Consensus (hard fork)
+  Title: Introduce new EncodeReveal RADON error
+  Authors: Tomasz Polaczyk <tomasz@witnet.foundation>
+  Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-01-12
+  License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+A new `EncodeReveal` RADON error is used as the result of a RADON aggregation script when the result cannot be encoded to CBOR.
+
+## Motivation and rationale
+
+There are some values that can be represented as RADON types, but cannot be represented using CBOR. For example, see this witnet request:
+
+```js
+const big_integer = new Witnet.Source("http://example.org/array_with_1_element.json")
+    .parseJSONArray()
+    .getInteger(0)
+    .multiply(1000000000000000)
+    .multiply(100000)
+```
+
+Where the source response is a JSON array with value `[1]`.
+
+The result of that retrieval script will be a RadonInteger greater than the maximum allowed by CBOR. This will not immediately result in an error, because the result of a retrieval script is not directly encoded to CBOR. However, if the aggregation script does not modify that value, by using a simple reducer like Mode, then the result of the aggregation script will be a RadonInteger greater than the maximum allowed by CBOR, and that will result in an encode error when trying to create the reveal transaction.
```

Comment:
```suggestion
The result of that retrieval script will be a `RadonInteger` greater than the maximum allowed by CBOR. This will not immediately result in an error, because the result of a retrieval script is not directly encoded to CBOR. However, if the aggregation script does not modify that value, by using a simple reducer like Mode, then the result of the aggregation script will be a `RadonInteger` greater than the maximum allowed by CBOR, and that will result in an encode error when trying to create the reveal transaction.
```